hpack/hpack.py

```diff
@@ -353,9 +354,10 @@ def decode(self, data, raw=False):
             current_index += consumed
 
         if raw:
-            return [(n, v) for n, v in headers]
+            return [(to_bytes(n), to_bytes(v)) for n, v in headers]
         else:
-            return [(n.decode('utf-8'), v.decode('utf-8')) for n, v in headers]
+            return [(to_bytes(n).decode('utf-8'), to_bytes(v).decode('utf-8'))
+                    for n, v in headers]
```

Comment:
Hmm, this section is getting a bit complex. Can we try to hoist some of this functionality out into a function or a lambda that would let us save some lines? For example, a function that takes a name-value tuple and the `raw` parameter, and returns it coded appropriately.